File: php/test/Behavioral/CommandTest.php

<?

require_once 'src/Behavioral/Command/DvdName.php';
require_once 'src/Behavioral/Command/DvdCommandNameStarsOn.php';
require_once 'src/Behavioral/Command/DvdCommandNameStarsOff.php';

class Behavioral_CommandTest extends PHPUnit_Framework_TestCase
{
	function test()
	{
		$jayAndBob = new DvdName("Jay and Silent Bob Strike Back");
		$spongeBob = new DvdName("Sponge Bob Squarepants - Nautical Nonsense and Sponge Buddies");

		$bobStarsOn = new DvdCommandNameStarsOn($jayAndBob);
		$bobStarsOff = new DvdCommandNameStarsOff($jayAndBob);
		$spongeStarsOn = new DvdCommandNameStarsOn($spongeBob);
		$spongeStarsOff = new DvdCommandNameStarsOff($spongeBob);

		$this->assertEquals($jayAndBob->toString(),'DVD: Jay and Silent Bob Strike Back','test to string');
		$this->assertEquals($spongeBob->toString(),'DVD: Sponge Bob Squarepants - Nautical Nonsense and Sponge Buddies','test to string');


		$bobStarsOn->execute();
		$spongeStarsOn->execute();
		$this->assertEquals($jayAndBob->toString(),'DVD: Jay*and*Silent*Bob*Strike*Back','test to string');
		$this->assertEquals($spongeBob->toString(),'DVD: Sponge*Bob*Squarepants*-*Nautical*Nonsense*and*Sponge*Buddies','test to string');

		$spongeStarsOff->execute();
		$this->assertEquals($jayAndBob->toString(),'DVD: Jay*and*Silent*Bob*Strike*Back','test to string');
		$this->assertEquals($spongeBob->toString(),'DVD: Sponge Bob Squarepants - Nautical Nonsense and Sponge Buddies','test to string');
	}
}

File: go/behavioral/command/command_test.go
package command

import (
	"testing"
	"../../asserts"
)

func TestCommand( t *testing.T) {            
	jayAndBob := NewDvdName("Jay and Silent Bob Strike Back")
	spongeBob := NewDvdName("Sponge Bob Squarepants - Nautical Nonsense and Sponge Buddies")

	bobStarsOn := NewDvdCommandNameStarsOn(jayAndBob)
	//bobStarsOff := NewDvdCommandNameStarsOff(jayAndBob)
	spongeStarsOn := NewDvdCommandNameStarsOn(spongeBob)
	spongeStarsOff := NewDvdCommandNameStarsOff(spongeBob)

	bobStarsOn.execute()
	spongeStarsOn.execute()

	asserts.Equals(t, "stars on jb test ", jayAndBob.ToString(),  "DVD: Jay*and*Silent*Bob*Strike*Back")
	asserts.Equals(t, "stars on sb test ", spongeBob.ToString(),  "DVD: Sponge*Bob*Squarepants*-*Nautical*Nonsense*and*Sponge*Buddies" )

	spongeStarsOff.execute()

	asserts.Equals(t, "sponge stars off test jb should be the same", jayAndBob.ToString(), "DVD: Jay*and*Silent*Bob*Strike*Back")
	asserts.Equals(t, "sponge stars off test sb should be changed", spongeBob.ToString(), "DVD: Sponge Bob Squarepants - Nautical Nonsense and Sponge Buddies")
}      
